qiskit/transpiler/passes/routing/swap_strategies/pauli_2q_evolution_commutation.py

```diff
@@ -0,0 +1,140 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An analysis pass to find evolution gates in which the Paulis commute."""
+
+from typing import Tuple
+
+import numpy as np
+
+from qiskit.exceptions import QiskitError
+from qiskit.circuit.library import PauliEvolutionGate
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.transpiler import TransformationPass
+from qiskit.quantum_info import SparsePauliOp, Pauli
+from qiskit.transpiler.passes.routing.swap_strategies.commuting_2q_block import Commuting2qBlocks
+
+
+class FindCommutingPauliEvolutions(TransformationPass):
```

Comment:
this pass would not find commuting CZ or ZZ gates if they are not specified using PauliEvolution, right? I think this limits its utility a bit. In general finding commuting gates should be a solved problem using the DAGDependency datastructure.

Reply:
We are working on a commutative analysis for this.

Reply:
Yes, currently we could use `qiskit.transpiler.passes.CommutationAnalysis` for this. As far as I understand this pass needs a bit of refactoring, then its output can be cast into the `Commuting2qBlocks` class that we have. This would allow us to extend to arbitrary gates. We do not use `CommutationAnalysis` for the `PauliEvolutionGate` as there are cheaper/faster ways to figure out if the gates commute.